File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heap = []

    for i in scoville:
        heapq.heappush(heap, i)
            
    while len(heap) > 1:
        cur = heapq.heappop(heap)
        if cur >= K:
            break
        cur2 = heapq.heappop(heap)

        heapq.heappush(heap, cur + (cur2*2))
        answer += 1
     
    if len(heap) != 0:
        if heapq.heappop(heap) < K:
            answer = -1
            
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> priorityQueue = new PriorityQueue<>();
        for(int i=0; i<scoville.length; i++){
            priorityQueue.add(scoville[i]);
        }
        
        while(priorityQueue.peek()<K){
            
        if(priorityQueue.size()==1){
             break;
         }

        int num = priorityQueue.poll();
        int num2 = priorityQueue.poll();
        
        int addNum = num + (num2 * 2);
        
        priorityQueue.add(addNum);
        answer++;
            
       } 
         if(priorityQueue.peek()<K){
             answer = -1;
         }
        return answer;
    }
}